src/BattleTest.java

```diff
@@ -0,0 +1,117 @@
+                public void	write(int b) {
+                    // noop
+                }
+            });
+    }
+
+    @Test
+    @DisplayName("Test for battle function")
+    public void testForBattle1() {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(90, 40, 30, 2, "Mike", "Mike story");
+
+        printerSetup();
+        Battle battle = new Battle(printer);
+
+        // //Act
+        battle.battle(entity1, entity2);
+
+        // Assert
+        assertTrue(entity1.getHealth() > entity2.getHealth(), "Wrong health");
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {90, 50, 70, 80, 40, 45.7, 27.9})
+    @DisplayName("Test for battle function 2")
+    public void testForBattle2(double health) {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(health, 40, 30, 2, "Mike", "Mike story");
+
+        printerSetup();
+        Battle battle = new Battle(printer);
+
+        //Act
+        battle.battle(entity1, entity2);
+
+        // Assert
+        assertTrue(entity1.isAlive(), "Wrong health");
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {17, 30, 50, 40, 15, 45.7, 27.9})
+    @DisplayName("Test for battle function 3")
+    public void testForBattle3(double attack) {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(90, attack, 30, 2, "Mike", "Mike story");
+
+        printerSetup();
+        Battle battle = new Battle(printer);
+
+        //Act
+        battle.battle(entity1, entity2);
+
+        // Assert
+        assertTrue(entity1.getHealth() > entity2.getHealth(), "Wrong health");
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {17, 30, 20, 30, 15, 15.7, 27.9})
+    @DisplayName("Test for battle function 4")
+    public void testForBattle4(double defense) {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(90, 40, defense, 2, "Mike", "Mike story");
+
+        printerSetup();
+        Battle battle = new Battle(printer);
+
+        //Act
+        battle.battle(entity1, entity2);
+
+        // Assert
+        assertTrue(entity1.isAlive(), "Wrong Parameters");
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {17, 30, 20, 30, 15, 15.7, 27.9})
+    @DisplayName("Test for battle function 5 - Test the winner")
+    public void testForBattle5(double defense) {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(90, 40, defense, 2, "Mike", "Mike story");
+
+        printerSetup();
+        Battle battle = new Battle(printer);
+
+        //Act
+        battle.battle(entity1, entity2);
+
+        Entity winner = entity1.isAlive() ? entity1 : entity2;
+        var result = winner.getName() + " won!";
```

Comment:
Ez teljesen felesleges. `Battle#battle`, mint korábban is mondtam, visszatéríti a harc kimenetelét.

Reply:
Most elvilegg jó, egyszerűbb és "szebb" lett. :)